pkg/controller/managedresources/controller.go

```diff
@@ -59,6 +60,29 @@ const (
 	FinalizerName = "resources.gardener.cloud/gardener-resource-manager"
 )
 
+const (
+	// Cluster Identity Handling
```

Comment:
Do we really need to support this feature that one can influence the algorithm of how the ID is determined, i.e. do we have any place today where we make use of this?

I think that users/operators might not know how to use it and except of this inline description, the feature is not mentioned in any user facing document.

Can't we simply assume that `cluser-id` is always passed as a flag?

Reply:
Not really, because the cluster-id is not available at the command line anymore in our scenarios. Here it should be taken from the cluster-id configmap in kube-system. That is exactly the reason, why I introduced these mechanisms. If someone is using the resource manager outside of a shoot cluster/garden environment, he should be able to explicitly specify or avoid a cluster id.

Reply:
Is there going to be a different scenario than today? Because when Gardener deploys the GRM as part of the control plane, the cluster id of the shoot is already available. The same is true for the GRM in the seed.